dspace/modules/additions/src/main/java/org/dspace/discovery/SolrServiceImpl.java

```diff
@@ -1009,6 +1011,55 @@ protected void buildDocument(Context context, Item item)
             }
         }
 
+        //TAMU Customization - Write bitstream URLs to index
+        List<String> bitstreamLocations = new ArrayList<>();
+        String dspaceUrl = ConfigurationManager.getProperty("dspace.url");
+        for (Bundle bundle : item.getBundles()) {
+            String bitstreamUrlTemplate = "%s/bitstream/handle/%s/%s?sequence=%d";
+            String primaryInternalId = null;
+            switch(bundle.getName()) {
+                case "ORIGINAL":
+                    if (bundle.getPrimaryBitstream() != null) {
+                        Bitstream primaryBitstream = bundle.getPrimaryBitstream();
+                        primaryInternalId = primaryBitstream.getInternalId();
+                        String primaryName = primaryBitstream.getName();
+                        int primarySequence = primaryBitstream.getSequenceID();
+                        String primaryUrl = String.format(bitstreamUrlTemplate, dspaceUrl, handle, primaryName, primarySequence);
+                        doc.addField("primaryBitstream", primaryUrl);
```

Comment:
Are you wanting the bitstream fields to be searchable and affect search results?

If not, you can add '_stored' as a suffix to the field names.

This will make the fields accessible as data to SAGE without affecting the actual SOLR searching.

That's what we did with the community/collection data:
https://github.com/TAMULib/DSpace/blob/0cbf0d1a17c9e343b521aa8f7edca91d71eb481e/dspace/modules/additions/src/main/java/org/dspace/discovery/SolrServiceImpl.java#L1010